scripts/release_functions.sh

```diff
@@ -1,12 +1,12 @@
 #!/usr/bin/env bash
 
 function find_gradle_property(){
+    [[ "$(declare -p $1 2>/dev/null)" =~ "declare -a" ]] || return 11
+    local -n output_target=$1
     for command_line_argument in "$@"; do
-        gradle_properties=""
         if [[ "${command_line_argument}" =~ ^-P.*=.* ]];then
-          gradle_properties="${gradle_properties} ${command_line_argument}"
+          output_target+=("${command_line_argument}")
```

Comment:
Just to get more insight: what is the point in constructing output_target? Where/how is it used?

Reply:
In a sense. What we are doing is that we pass the name of the actual variable to the function. Then we create a reference (like an always associated pointer) to that variable with declare -n. And then we address the actual variable by its reference in the function. 

Since bash functions cannot really return complex information, this is the best way to isolate functionality.